XenAdmin/Dialogs/AttachUsbDialog.cs

```diff
@@ -56,16 +56,37 @@ private void BuildList()
             treeUsbList.BeginUpdate();
             try
             {
-                Host[] hosts = _vm.Connection.Cache.Hosts;
-                foreach (Host host in hosts)
+                VM.HA_Restart_Priority SelectedPriority = _vm.HARestartPriority();
+                Pool pool = Helpers.GetPool(_vm.Connection);
+                // Check if HA was enabled on pool and Restart priority was set on VM.
+                bool haEnabled = (pool != null &&

```

Comment:
The HA check should be done on the form that opens this dialog, where we should disable the USB attach button and display the message stating why cannot attach USB devices.

Reply:
Good idea. HA-check is on VM, but not on each candidate, should be done on the form. Accepted.